sdks/go/test/load/sideinput/sideinput.go

```diff
@@ -0,0 +1,100 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"reflect"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/io/synthetic"
+	"github.com/apache/beam/sdks/go/pkg/beam/log"
+	"github.com/apache/beam/sdks/go/pkg/beam/x/beamx"
+	"github.com/apache/beam/sdks/go/test/load"
+)
+
+func init() {
+	beam.RegisterDoFn(reflect.TypeOf((*doFn)(nil)))
+}
+
+var (
+	accessPercentage = flag.Int(
+		"access_percentage",
+		100,
+		"Specifies the percentage of elements in the side input to be accessed.")
+	syntheticSourceConfig = flag.String(
+		"input_options",
+		"",
+		"A JSON object that describes the configuration for synthetic source")
+)
+
+func parseSyntheticConfig() synthetic.SourceConfig {
+	if *syntheticSourceConfig == "" {
+		panic("--input_options not provided")
+	} else {
+		encoded := []byte(*syntheticSourceConfig)
+		return synthetic.DefaultSourceConfig().BuildFromJSON(encoded)
+	}
+}
+
+type doFn struct {
+	elementsToAccess int
```

Comment:
This isn't the main problem, we're debugging, but DoFn fields *must* be exported to get serialized properly.

eg. ElementsToAccess instead.

Reply:
Thank you, I changed the name to start with capital letter.